prefect_dask/task_runners.py

```diff
@@ -276,18 +276,24 @@ async def submit(
         else:
             dask_key = key
 
-        self._dask_futures[key] = self._client.submit(
-            call.func,
-            key=dask_key,
-            # Dask defaults to treating functions are pure, but we set this here for
-            # explicit expectations. If this task run is submitted to Dask twice, the
-            # result of the first run should be returned. Subsequent runs would return
-            # `Abort` exceptions if they were submitted again.
-            pure=True,
-            **call_kwargs,
+        self._dask_futures[key] = distributed.as_completed(
+            futures=[
+                self._client.submit(
+                    call.func,
+                    key=dask_key,
+                    # Dask defaults to treating functions are pure, but we set this here
+                    # for explicit expectations. If this task run is submitted to Dask
+                    # twice, the result of the first run should be returned. Subsequent
+                    # runs would return `Abort` exceptions if they were submitted again.
+                    pure=True,
+                    **call_kwargs,
+                )
+            ],
+            loop=self._client.loop,
+            with_results=True,
```

Comment:
This line does the trick to have the expected behavior (no completed tasks retry) but this means that the results are fetched inside the `as_completed` class so the timeout can't be passed in the `wait` method.